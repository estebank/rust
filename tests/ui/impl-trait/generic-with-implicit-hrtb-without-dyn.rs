//@ revisions: edition2015 edition2021
//@[edition2021]edition:2021

#![allow(warnings)]

fn ice() -> impl AsRef<Fn(&())> {
    //[edition2015]~^ ERROR trait `AsRef<dyn for<'a> Fn(&'a ())>` is not implemented for `()`
    //[edition2021]~^^ ERROR: trait objects must include the `dyn` keyword [E0782]
    //[edition2021]~| ERROR trait `AsRef<dyn for<'a> Fn(&'a ())>` is not implemented for `()`
    todo!()
}

fn main() {}
