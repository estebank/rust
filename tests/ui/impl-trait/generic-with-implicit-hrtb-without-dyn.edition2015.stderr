error[E0277]: the trait `AsRef<dyn for<'a> Fn(&'a ())>` is not implemented for `()`
  --> $DIR/generic-with-implicit-hrtb-without-dyn.rs:6:13
   |
LL | fn ice() -> impl AsRef<Fn(&())> {
   |             ^^^^^^^^^^^^^^^^^^^ the trait `AsRef<dyn for<'a> Fn(&'a ())>` is not implemented for `()`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
